# Block 1

In [3]:
# Import pandas and files
import pandas as pd
from google.colab import files

# Block 2

In [4]:
# Load in the data
files.upload()
data = pd.read_csv('problem_set_1_data.csv')

# Rename the columns to be “quiz_total”, “attended”, “final_grade”, “final_exam”, and “name”
data.columns = ["quiz_total", "attended", "final_grade", "final_exam", "name"]

# Print the head of the dataframe
print()
print("Head of dataframe:")
print(data.head())
print()

# Print the tail of the dataframe
print("Tail of dataframe:")
print(data.tail())
print()

# Print a summary of the dataframe
print("Summary of the dataframe:")
print(data.describe())

Saving problem_set_1_data.csv to problem_set_1_data.csv

Head of dataframe:
   quiz_total  attended  final_grade  final_exam               name
0       78.50         1        99.38          46       Linnie Lietz
1       64.25         0        95.78          45      Mitch Mustain
2       76.75         0        97.42          41     Salina Chavera
3       69.00         1        92.50          38  Kimberely Conwell
4       64.00         0        92.50          39          Zack Burk

Tail of dataframe:
     quiz_total  attended  final_grade  final_exam                 name
166        73.5         0        96.25          45           Dana Malta
167        75.5         0        98.13          44         Sixta Heyden
168        41.0         0        69.38          32       Brianne Broome
169        77.0         0       101.25          47  Elouise Weatherholt
170        72.0         0        92.81          38        Addie Maharaj

Summary of the dataframe:
       quiz_total    attended  final_

The mean for each column displays the average within each column. This is calculated by taking the sum of all values in the column and dividing by the number of observations. For example, the mean for quiz total, 69.13 represents the average quiz total for all observations. This is the same for the final grade and final exam variables. The mean for attended is somewhat different because it is a dichotomous variable (only two options). For this variable, the mean represents the percentage of observations that have a value of 1. Since 1 correspons to attending class in this dataset, the mean of .19 signifies that 19% of students attended class.

# Block 3

In [11]:
# Subset the data by whether someone attended class
attended = data[data["attended"] == 1]
did_not_attend = data[data["attended"] == 0]

# Find the average quiz total, final grade, and final exam for students who attended class
print("Average quiz total for students who attended class:", attended['quiz_total'].mean())
print("Average final grade for students who attended class:", attended['final_grade'].mean())
print("Average final exam for students who attended class:", attended['final_exam'].mean())
print()

# Find the average of the same variables for students who did not attend class
print("Average quiz total for students who did not attend class:", did_not_attend['quiz_total'].mean())
print("Average final grade for students who did not attend class:", did_not_attend['final_grade'].mean())
print("Average final exam for students who did not attend class:", did_not_attend['final_exam'].mean())

Average quiz total for students who attended class: 72.5
Average final grade for students who attended class: 95.07212121212123
Average final exam for students who attended class: 40.63636363636363

Average quiz total for students who did not attend class: 68.32608695652173
Average final grade for students who did not attend class: 91.08615942028983
Average final exam for students who did not attend class: 39.44202898550725


When comparing the average quiz total, final grade, and final exam scores between students who attended class and students who did not attend class, we see that, on average, students who attended class received better quiz scores, final exam scores, and final grades than students who did not attend class.
* On average, students who attended class received 4.17 (72.5 - 68.33) more points on their quiz total than students who did not attend class.
* On average, students who attended class had 3.98 (95.07 - 91.09) more percentage points on their final grade than students who did not attend class.
* On average, students who attended class received 1.2 (40.64 - 39.44) more points on their final exam than students who did not attend class.


# Block 4

In [44]:
data[data.duplicated(subset = ["quiz_total", "final_exam", "final_grade"], keep = False)]

quiz_total  attended  final_grade  final_exam           name
146        72.0         0        92.81          38  Jaraham Eidda
170        72.0         0        92.81          38  Addie Maharaj